### PR TITLE
Autocompletion for Class Code Text Box

### PR DESCRIPTION
**Pull request by NiharikaVadlamudi**

### CODE REVIEW

File: Attendance_app.py [hunk 1]
```diff
@@ -185,9 +191,22 @@ def __init__(self, parent, controller):
...
+        self.classCodes=list(class_codes)
+        self.class_variable = StringVar(self)
+        self.class_variable.set('') # default value
+        self.tv_class=tk.Entry(self,textvariable=self.class_variable)
+        self.tv_class.focus_set()
+        self.tv_class.bind('<KeyRelease>', self.get_typed)
+        self.tv_class.bind('<Key>', self.detect_pressed)
+        self.tv_class.place(x=400* scale_factor, y=160* scale_factor)
+
+
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py:209]: Fix [PEP8](https://www.python.org/dev/peps/pep-0008/) Issues:
```
./Attendance_app.py:196:1: W293 blank line contains whitespace
./Attendance_app.py:200:24: E225 missing whitespace around operator
./Attendance_app.py:201:31: F405 'StringVar' may be undefined,
                                  or defined from star imports: tkinter
./Attendance_app.py:202:36: E261 at least two spaces before inline comment
./Attendance_app.py:203:22: E225 missing whitespace around operator
./Attendance_app.py:203:36: E231 missing whitespace after ','
./Attendance_app.py:207:34: E225 missing whitespace around operator
./Attendance_app.py:207:55: E225 missing whitespace around operator
./Attendance_app.py:209:1: W293 blank line contains whitespace
```

File: Attendance_app.py [hunk 2]
```diff
@@ -249,6 +268,33 @@ def doWork(self, page_name, c_code, id, pwd):
...
+                self.class_variable.set(lst[0])
+                self.filled = True
+
+    def detect_pressed(self,event):    
+            self.key = event.keysym
+            if len(self.key) == 1 and self.filled is True:
+                pos = self.tv_class.index(tk.INSERT)
+                self.tv_class.delete(pos, tk.END)
+
+
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py:299]: Fix [PEP8](https://www.python.org/dev/peps/pep-0008/) Issues:
```
./Attendance_app.py:271:1: W293 blank line contains whitespace
./Attendance_app.py:272:36: W291 trailing whitespace
./Attendance_app.py:274:13: E117 over-indented
./Attendance_app.py:279:24: W291 trailing whitespace
./Attendance_app.py:281:23: E231 missing whitespace after ','
./Attendance_app.py:282:13: E117 over-indented
./Attendance_app.py:283:23: E225 missing whitespace around operator
./Attendance_app.py:286:22: E231 missing whitespace after ','
./Attendance_app.py:287:13: E117 over-indented
./Attendance_app.py:291:28: E231 missing whitespace after ','
./Attendance_app.py:291:36: W291 trailing whitespace
./Attendance_app.py:292:13: E117 over-indented
./Attendance_app.py:301:1: E303 too many blank lines (4)
```

File: Attendance_app.py [hunk 3]
```diff
@@ -17,6 +17,12 @@
 from students_list import StudentsList
 from main_file import MainFile
 
+import tkinter as tk
+import tkinter.filedialog as tkfd
+from tkinter import *
+from tkinter import ttk
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py:23]: Fix [PEP8](https://www.python.org/dev/peps/pep-0008/) Issues:
```
./Attendance_app.py:20:1: F811 redefinition of unused 'tk' from line 5
./Attendance_app.py:21:1: F401 'tkinter.filedialog as tkfd' imported but unused
./Attendance_app.py:22:1: F403 'from tkinter import *' used;
                                unable to detect undefined names
./Attendance_app.py:23:1: F401 'tkinter.ttk' imported but unused
./Attendance_app.py:23:24: W291 trailing whitespace
```

File: Attendance_app.py [hunk 4]
```diff
@@ -185,9 +191,22 @@ def __init__(self, parent, controller):
                                     fg=self.text_color,
                                     font=("Courier", text_fontsize))
         self.lb_username.place(x=183 * scale_factor, y=210 * scale_factor)
-        self.tv_username = tk.Entry(self, width=entry_width)
-        self.tv_username.place(x=340 * scale_factor, y=215 * scale_factor)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py:189]: Accidentally deleted these lines?

File: Attendance_app.py [hunk 5]
```diff
@@ -185,9 +191,22 @@ def __init__(self, parent, controller):
...
+        self.filled = False
+
+        self.classCodes=list(class_codes)
+        self.class_variable = StringVar(self)
+        self.class_variable.set('') # default value
+        self.tv_class=tk.Entry(self,textvariable=self.class_variable)
+        self.tv_class.focus_set()
+        self.tv_class.bind('<KeyRelease>', self.get_typed)
+        self.tv_class.bind('<Key>', self.detect_pressed)
+        self.tv_class.place(x=400* scale_factor, y=160* scale_factor)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py:207]: Initialization, focussing and placement have been done twice.
Also done on these lines:
https://github.com/Marauders-9998/Attendance-Management-using-Face-Recognition/blob/866f0cf26a3b5b11f0c13d4c08bc211f2e8feef5/Attendance_app.py#L186-L188

**Review by Vibhu-Agarwal — CHANGES_REQUESTED** [review]
AMFR_auto_complete_bug

Found these issues:
- Two and overlapping text-boxes for the class-code
  - one, just like the current one
  - other, new one, with auto-completion
- Text-box for entering Username has been removed

Issues, specifically related to auto-completion:
- After the auto-completion, the cursor remains at the point till where we typed.
  Ideally, auto-completion in CLI tools happens on the press of a `Tab` and we always expect the cursor to be at the end of the last character.
  This is highly opinionated though. If you agree with my point, you can make appropriate changes.

And rest of the issues have been pointed out in comments.

**Comment by NiharikaVadlamudi** [comment]
@Vibhu-Agarwal  Pushed the wrong file before, apologies for that, couldn't fix the cursor issue, but I do agree with your suggestions on how it should function.

**Review by Vibhu-Agarwal — CHANGES_REQUESTED** [review]
@NiharikaVadlamudi Please go to the [`File changed`](https://github.com/Marauders-9998/Attendance-Management-using-Face-Recognition/pull/29/files) tab and self-review the PR once.
It seems that the whole file is changed. I'm unable to spot the exact (and intended) changes made in this PR.

Hint: It seems you're using a different code-formatter than the repository's [prefrred one](https://black.readthedocs.io/en/stable/).
The README also has a badge for it ;)

**Pull request closed without merging**